src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -267,4 +282,60 @@ case class CreateDeltaTableCommand(
       }
     }
   }
+
+  private def getOperation(
+      tableVersion: Long,
+      metadata: Metadata,
+      isManagedTable: Boolean,
+      options: Option[DeltaOptions]): DeltaOperations.Operation = operation match {
+    case TableCreationModes.CreateTable if existingTableOpt.isDefined && query.isDefined =>
+      DeltaOperations.Write(mode, Option(table.partitionColumnNames), options.get.replaceWhere)
+
+    case TableCreationModes.CreateTable =>
+      DeltaOperations.CreateTable(metadata, isManagedTable, query.isDefined)
+
+    case TableCreationModes.ReplaceTable =>
+      DeltaOperations.ReplaceTable(metadata, isManagedTable, orCreate = false, query.isDefined)
+
+    case TableCreationModes.CreateOrReplaceTable if options.exists(_.replaceWhere.isDefined) =>
+      DeltaOperations.Write(mode, Option(table.partitionColumnNames), options.get.replaceWhere)
+
+    case TableCreationModes.CreateOrReplaceTable =>
+      DeltaOperations.ReplaceTable(metadata, isManagedTable, orCreate = true, query.isDefined)
+  }
+
+  private def updateCatalog(spark: SparkSession, table: CatalogTable): Unit = operation match {
+    case TableCreationModes.CreateTable =>
+      spark.sessionState.catalog.createTable(
+        table,
+        ignoreIfExists = existingTableOpt.isDefined,
+        validateLocation = false)
+    case TableCreationModes.ReplaceTable if existingTableOpt.isDefined =>
+      spark.sessionState.catalog.alterTable(table)
```

Comment:
Should this be alter or drop and recreate? (I'm wondering whether there are Hive-side constraints on what set of alterations are allowed.)

Reply:
that's a great question. Since we're only storing location in the HiveMetaStore, I think this should be safe.